dvc/remote/local.py

```diff
@@ -49,10 +58,24 @@ def __init__(self, repo, config):
         super().__init__(repo, config)
         self.cache_dir = config.get("url")
         self._dir_info = {}
+        # repo.tree can be WorkingTree or GitTree, but GitTree does not contain
+        # cache directory
+        if self.repo:
+            if is_working_tree(self.repo.tree):
+                self._tree = self.repo.tree
+            else:
+                tree = WorkingTree(self.repo.root_dir)
+                if isinstance(self.repo.tree, CleanTree):
+                    tree = CleanTree(tree)
```

Comment:
I am not sure I follow here.
We get to this `else` statement if `repo.tree` is not `working_tree`, this happens when we are playing around with brancher.
So why do we need information whether tree on some branch is CleanTree?

Reply:
`repo.tree` will be `CleanTree(GitTree())` for external dvc repos, so I think my original logic here was that the new `WorkingTree` should also be wrapped with `CleanTree`.

But looking at it again, you're right. We only need `tree = WorkingTree(...)` to be able to see the .dvc/cache directory in this case, but it doesn't matter whether or not we wrap it with CleanTree.

Reply:
fixed in d010f7f